x-pack/plugins/cases/public/components/markdown_editor/use_markdown_session_storage.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useRef, useState } from 'react';
+import { isEmpty } from 'lodash';
+import useDebounce from 'react-use/lib/useDebounce';
+import useSessionStorage from 'react-use/lib/useSessionStorage';
+import type { FieldHook } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+
+const STORAGE_DEBOUNCE_TIME = 500;
+
+interface SessionStorageType {
+  field: FieldHook<string>;
+  sessionKey: string;
+  initialValue: string | undefined;
+}
+
+export const useMarkdownSessionStorage = ({
+  field,
+  sessionKey,
+  initialValue,
+}: SessionStorageType) => {
+  const [hasConflicts, setHasConflicts] = useState<boolean>(false);
+  const isFirstRender = useRef(true);
+  const initialValueRef = useRef(initialValue);
+
+  const [sessionValue, setSessionValue] = useSessionStorage<string>(sessionKey);
+
+  if (!isEmpty(sessionValue) && isFirstRender.current) {
+    isFirstRender.current = false;
+    field.setValue(sessionValue);
+  }
+
+  if (initialValue !== initialValueRef.current && initialValue !== field.value) {
+    initialValueRef.current = initialValue;
+    setHasConflicts(true);
+  }
+
+  useDebounce(
+    () => {
+      if (isFirstRender.current) {
+        isFirstRender.current = false;
+        return;
+      }
+
+      if (sessionKey) {
+        if (field.value !== '') {
```

Comment:
As discussed, sometimes when I am editing a case across two tabs, I see the draft comment added in a new comment after saving the original changes.

Here on saving the new comment `Adding a comment in tab 2`, a new draft section was added below with the same value:

<img width="625" alt="image" src="https://user-images.githubusercontent.com/7405507/207023172-f584b753-44e8-4ccc-a91f-7d4b82bf94ab.png">

Reply:
Fixed this issue, and verified locally.
Can you please also check once? @peteharverson

Reply:
I am still seeing this occasionally, but seems to happen less often now - maybe once every 10 edits. If you are able to reproduce, I'd suggest looking at it in a separate PR rather than holding up this PR as it's a rare edge case.